generators/typescript-v2/ast/src/ast/Func.ts

```diff
@@ -0,0 +1,89 @@
+import { CodeBlock } from "./CodeBlock";
+import { Comment } from "./Comment";
+import { Parameter } from "./Parameter";
+import { Type } from "./Type";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+
+export declare namespace Func {
+    interface Args {
+        name: string;
+        parameters: Parameter[];
+        return_: Type[];
+        body: CodeBlock | undefined;
+        docs: string | undefined;
+    }
+}
+
+export class Func extends AstNode {
+    protected _name: string;
+    protected _parameters: Parameter[];
+    protected _return_: Type[];
+    protected _body: CodeBlock | undefined;
+    protected _docs: string | undefined;
+
+    constructor({ name, parameters, return_, body, docs }: Func.Args) {
+        super();
+        this._name = name;
+        this._parameters = parameters;
+        this._return_ = return_;
+        this._body = body;
+        this._docs = docs;
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public get parameters(): Parameter[] {
+        return this._parameters;
+    }
+
+    public get return_(): Type[] {
+        return this._return_;
+    }
+
+    public get body(): CodeBlock | undefined {
+        return this._body;
+    }
+
+    public get docs(): string | undefined {
+        return this._docs;
+    }
+
+    public write(writer: Writer): void {
+        writer.writeNode(new Comment({ docs: this.docs }));
+        writer.write("function ");
+        writer.write(`${this.name}`);
+        writer.write("(");
+        writer.delimit({
+            nodes: this._parameters,
+            delimiter: ",\n",
```

Comment:
This feels off - including the `\n` within the delimiter feels a little bug prone, especially because we have the `writer.writeLine` method already.

With that said, it might be simpler just to write the implementation more explicitly in each call-site, even if you end up with the same shape of code across different AST nodes.

We can leave this as-is for now and come back to it though.

Reply:
yeah, true. I'm going to mull it over for a bit

Reply:
This could be an option inside the delimit function telling it each item has to be on a new line.

Reply:
@Swimburger Agreed, I actually had the same thought [here](https://github.com/fern-api/fern/pulls#discussion_r1887542602).